src/main/webapp/facebook-login.js

```diff
@@ -21,35 +21,40 @@ function include(javascriptFile) {
   script.defer = true; 
   document.getElementsByTagName('head').item(0).appendChild(script); 
 } 
-include('./FacebookAPI.js');
 include('./load-facebook.js');
-
+ 
 // Response to the user's login status, redirect appropriately
 function statusChangeCallback(response) {
   if (response.status === 'connected') {
     // User is logged in with Facebook
     // TODO: Redirect user to the correct page depending on whether their profile is complete
     // For now, redirect to the profile page
     // Query Facebook Graph API for user information
-    FacebookAPI.getUserFacebookInfoAndRedirect(
-      function(response) {
-        let userInfo = new URLSearchParams({
-          name: response.name,
-          id: response.id,
-          email: response.email,
-        });
-
-        // Store user information in Datastore
-        // TODO: Add post request
-        // Redirect to the profile page, and pass the user's info in the URL
+    FB.api('/me?fields=id,name,email,friends', function(response) {
+      let userInfo = new URLSearchParams({
+        name: response.name,
+        id: response.id,
+        email: response.email,
+      });
+ 
+      // Add friend ids to the query string
+      // TODO: (After MVP) Replace this with jquery + ajax to POST JSON to the backend
+      response.friends.data.forEach((friend) => userInfo.append("friends-list", friend.id));
+ 
+      // Store user information in Datastore
+      fetch('/user-data?' + userInfo.toString(), {
+        method: 'POST',
+      });
+ 
+      // Redirect to the profile page, and pass the user's info in the URL
       window.location.href = "./profile.html?" + userInfo.toString();
     });
   }
 }
-
+ 
 // Called after the user logs in with Facebook
 function checkLoginState() {
-  FacebookAPI.checkLoginStatus(function(response) {
+  FB.getLoginStatus(function(response) {
     statusChangeCallback(response);
   });
 }
```

Comment:
nit: new line at end of file

Reply:
Thanks! Good catch.

Reply:
That's weird, it looks like there is a newline in my code, but not on github. I'll add an extra newline and see if that fixes the problem.

Reply:
Hmm... there seems to be some bug. Whenever I 'edit' the file on github, there appears to be 2 newlines. However, when I view the file on github, there are no newlines. On my local cloud shell, there is one new line, and git says no new changes are available to commit.

Reply:
Not sure what's going on here, I'm leaving this be for now. I think that this can also be fixed when merging (or in another branch).